src/workerd/api/sockets.h

```diff
@@ -55,10 +69,11 @@ class Socket: public jsg::Object {
   kj::Own<jsg::PromiseResolverPair<void>> closeFulfiller;
   // This fulfiller is used to resolve the `closedPromise` below.
   jsg::MemoizedIdentity<jsg::Promise<void>> closedPromise;
+  IoOwn<jsg::Promise<void>> writeDisconnectedPromise;
```

Comment:
`IoOwn` is only needed to wrap a `kj::Promise`. If this is a `jsg::Promise` then `IoOwn` is not needed.